src/data/ha-state-history-data.js

```diff
@@ -28,8 +28,12 @@ function computeHistory(stateHistory, localize, language) {
 
     const stateWithUnit = stateInfo.find(state => 'unit_of_measurement' in state.attributes);
 
-    const unit = stateWithUnit ?
-      stateWithUnit.attributes.unit_of_measurement : false;
+    let unit = false;
+    if (stateWithUnit) {
+      unit = stateWithUnit.attributes.unit_of_measurement;
+    } else if (stateInfo.some(state => computeStateDomain(state) === 'climate')) {
+      unit = hass.config.core.unit_system.temperature;
```

Comment:
Is hass.config.core.unit_system.temperature always there? Or might we get an exception here?

Reply:
It should always be there. I think right now we don't enforce it but we should.